python/pyspark/ml/__init__.py

```diff
@@ -21,5 +21,11 @@
 """
 from pyspark.ml.base import Estimator, Model, Transformer, UnaryTransformer
 from pyspark.ml.pipeline import Pipeline, PipelineModel
+from pyspark.ml import classification, clustering, evaluation, feature, fpm, \
+    image, pipeline, recommendation, regression, stat, tuning, util
 
-__all__ = ["Transformer", "UnaryTransformer", "Estimator", "Model", "Pipeline", "PipelineModel"]
+__all__ = [
+    "Transformer", "UnaryTransformer", "Estimator", "Model", "Pipeline", "PipelineModel",
+    "base", "classification", "clustering", "evaluation", "feature", "fpm",
+    "image", "pipeline", "recommendation", "regression", "stat", "tuning", "util",
```

Comment:
we should remove `pipeline` from default imports and add `linalg` and `param`. Basically, import public modules listed here: https://spark.apache.org/docs/latest/api/python/index.html.